src/preload/preload.c

```diff
@@ -685,7 +703,7 @@ static void __attribute__((constructor)) init_process(void) {
 
   buffer_enabled = !!getenv(SYSCALLBUF_ENABLED_ENV_VAR);
 
-  pthread_atfork(NULL, NULL, post_fork_child);
+  signal(SYSCALLBUF_DESCHED_SIGNAL, sig_init_thread);
```

Comment:
I don't think this is actually the case.  Note that this is before we call `rrcall_init_preload`, so rr doesn't know the syscallbuf is enabled yet, and `RecordTask::is_in_syscallbuf()` will return false anyways.  Also it would be safe to deliver signals in a traced syscall.  The only possible behavior difference I can see is that not using the rr page could make us skip `fixup_syscall_registers` at the end of `Task::did_waitpid`.  But that's also for untraced syscalls.

This is also a bit of a PITA to fix, because the libc wrapper sets up the restore trampoline for us, so if we can avoid this it would be nice.